components/brave_wallet/browser/zcash/zcash_blocks_batch_scan_task.cc

```diff
@@ -233,30 +237,8 @@ void ZCashBlocksBatchScanTask::OnBlocksScanned(
   ScheduleWorkOnTask();
 }
 
-void ZCashBlocksBatchScanTask::UpdateDatabase() {
-  CHECK(scan_result_.has_value());
-  CHECK(latest_scanned_block_.has_value());
-  auto latest_scanned_block_hash = ToHex((*latest_scanned_block_)->hash);
-  auto latest_scanned_block_height = (*latest_scanned_block_)->height;
-
-  context_->sync_state->AsyncCall(&OrchardSyncState::ApplyScanResults)
-      .WithArgs(context_->account_id.Clone(), std::move(scan_result_.value()),
-                latest_scanned_block_height, latest_scanned_block_hash)
-      .Then(base::BindOnce(&ZCashBlocksBatchScanTask::OnDatabaseUpdated,
-                           weak_ptr_factory_.GetWeakPtr()));
-}
-
-void ZCashBlocksBatchScanTask::OnDatabaseUpdated(
-    base::expected<OrchardStorage::Result, OrchardStorage::Error> result) {
-  if (!result.has_value()) {
-    error_ = ZCashShieldSyncService::Error{
-        ZCashShieldSyncService::ErrorCode::kFailedToUpdateDatabase,
-        result.error().message};
-    ScheduleWorkOnTask();
-    return;
-  }
-  database_updated_ = true;
-  ScheduleWorkOnTask();
+OrchardBlockScanner::Result ZCashBlocksBatchScanTask::TakeResult() {
+  return std::move(scan_result_.value());
 }
```

Comment:
would that make sense to guard this somehow so `TakeResult()` is called once?

Reply:
Will do in follow up